### PR TITLE
Update Kafka module to comply with Ballerina 0.991.0 release

### PR DESCRIPTION
**Pull request by ThisaruGuruge**

### CODE REVIEW

File: component/src/test/resources/consumer/kafka_consumer_subscribe_to_pattern.bal [hunk 1]
```diff
@@ -33,11 +33,19 @@ function funcKafkaTestSubscribeToPattern(kafka:SimpleConsumer kafkaConsumer) {
 }
 
 function funcKafkaTestGetSubscribedTopicCount(kafka:SimpleConsumer kafkaConsumer) returns int|error {
-    string[] subscribedTopics = check kafkaConsumer->getSubscription();
-    return (subscribedTopics.length());
+    var result = kafkaConsumer->getSubscription();
+    if (result is error) {
+        return result;
+    } else {
+        return (result.length());
```
Comment (wggihan) [inline component/src/test/resources/consumer/kafka_consumer_subscribe_to_pattern.bal]:
```suggestion
        return result.length();
```

File: component/src/test/java/org/ballerinalang/kafka/test/utils/Utils.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package org.ballerinalang.kafka.test.utils;
+
+public class Utils {
```
Comment (riyafa) [inline component/src/test/java/org/ballerinalang/kafka/test/utils/Utils.java]: Shall we rename to `Constants` if it is the Constants' class

File: component/src/main/wso2/kafka/consumer.bal [hunk 3]
```diff
@@ -141,169 +139,169 @@ public type SimpleConsumer client object {
         return self.stop();
     }
 
-    public function __attach(service s, map<any> annotationData) returns error? {
-        return self.registerListener(s, annotationData);
+    public function __attach(service s, string? name) returns error? {
```
Comment (MaryamZi) [inline component/src/main/wso2/kafka/consumer.bal]: `name` should be a defaultable param right? https://github.com/ballerina-platform/ballerina-lang/blob/master/stdlib/builtin/src/main/ballerina/builtin/listener.bal#L23
```suggestion
    public function __attach(service s, string? name = ()) returns error? {
```

This should ideally have failed at compile time, but doesn't at the moment due to a bug.

File: component/src/test/java/org/ballerinalang/kafka/test/utils/Constants.java [hunk 4]
```diff
@@ -0,0 +1,6 @@
+package org.ballerinalang.kafka.test.utils;
```
Comment (MaryamZi) [inline component/src/test/java/org/ballerinalang/kafka/test/utils/Constants.java]: Missing licence header?

File: component/src/test/resources/consumer/kafka_consumer.bal [hunk 5]
```diff
@@ -36,31 +36,26 @@ function funcKafkaConnectNegative() returns error? {
...
 
 function funcKafkaClose(kafka:SimpleConsumer consumer) returns boolean {
-    kafka:SimpleConsumer consumerEP = consumer;
-    var conErr = consumerEP->close();
-    if (conErr is error) {
+    var result = consumer->close();
+    if (result is error) {
         return false;
     }
     return true;
```
Comment (MaryamZi) [inline component/src/test/resources/consumer/kafka_consumer.bal]: Shall we replace L40-L43 with 
```ballerina
    return !(result is error);
```
if it doesn't make the code less readable?

**Pull request merged**